### ISSUE TITLE
Fields not registered in Items.fields if set as static variable

### ISSUE THREAD

**Issue by scratchmex** [scrapy/scrapy#4276]
### Description

The static variables set in `scrapy.Items` class are not registered in the internal `_values` dict and therefore I cannot access them via `self['field_name']`, i.e., the `__getattr__` function.

### Steps to Reproduce
In my items_error.py file I have this. (the actual code is larger but I isolate the code causing the error)
```
from scrapy import Item, Field

class BaseItem(Item):
    url=Field()
    id=Field()
    type=Field()

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        if not self['type']: #<--- error here. Raises KeyError
            raise NotImplementedError('type of ad must be set as static class variable')
        #use url hash as id to avoid duplicates
        self['id']=hash(self.get('url'))

class ChildItem(BaseItem):
    type='estate'
    other_fields=Field()
```

When I ran it this raises:
```
C:\Users\ivangonzalez\Documents\Code\scrady>python
Python 3.7.6 | packaged by conda-forge | (default, Jan  7 2020, 21:48:41) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from scrady.items_error import *
>>> ChildItem()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\ivangonzalez\Documents\Code\scrady\scrady\items_error.py", line 10, in __init__
    if not self['type']: #<--- error here. Raises KeyError
  File "C:\Users\ivangonzalez\Anaconda3\lib\site-packages\scrapy\item.py", line 91, in __getitem__
    return self._values[key]
KeyError: 'type'
>>> BaseItem()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\ivangonzalez\Documents\Code\scrady\scrady\items_error.py", line 10, in __init__
    if not self['type']: #<--- error here. Raises KeyError
  File "C:\Users\ivangonzalez\.virtualenvs\scrady-d4yjsu4b\lib\site-packages\scrapy\item.py", line 91, in __getitem__       return self._values[key]
KeyError: 'type'
>>>
```

**Expected behavior:** `self['field_name']` should return the `Field()` attribute.

**Actual behavior:** static fields are not registered by `scrapy.Items`  in `scrapy.Items._values[key]`.

**Reproduces how often:** Ran in pipenv, so 100%.

### Versions

Please paste here the output of executing `scrapy version --verbose` in the command line:
```
C:\Users\ivangonzalez\Documents\Code\scrady>scrapy version --verbose
Scrapy       : 1.8.0
lxml         : 4.4.2.0
libxml2      : 2.9.5
cssselect    : 1.1.0
parsel       : 1.5.2
w3lib        : 1.21.0
Twisted      : 19.10.0
Python       : 3.7.6 | packaged by conda-forge | (default, Jan  7 2020, 21:48:41) [MSC v.1916 64 bit (AMD64)]
pyOpenSSL    : 19.1.0 (OpenSSL 1.1.1d  10 Sep 2019)
cryptography : 2.8
Platform     : Windows-10-10.0.18362-SP0
```

### Additional context

In scrapy/item.py : ItemMeta class the attributes should be registered by this code:

```
class ItemMeta(ABCMeta):
    """Metaclass_ of :class:`Item` that handles field definitions.

    .. _metaclass: https://realpython.com/python-metaclasses
    """

    def __new__(mcs, class_name, bases, attrs):
        classcell = attrs.pop('__classcell__', None)
        new_bases = tuple(base._class for base in bases if hasattr(base, '_class'))
        _class = super(ItemMeta, mcs).__new__(mcs, 'x_' + class_name, new_bases, attrs)

        fields = getattr(_class, 'fields', {})
        new_attrs = {}
        for n in dir(_class):
            v = getattr(_class, n)
            if isinstance(v, Field):
                fields[n] = v
            elif n in attrs:
                new_attrs[n] = attrs[n] #<--- here

        new_attrs['fields'] = fields
        new_attrs['_class'] = _class
        if classcell is not None:
            new_attrs['__classcell__'] = classcell
        return super(ItemMeta, mcs).__new__(mcs, class_name, bases, new_attrs)
```

But they aren't. I don't know why.

**Comment by Gallaecio** [comment]
Have you considered using `self.type` instead of `self['type']`?

**Comment by scratchmex** [comment]
The class itself don't allow it:

```
    raise AttributeError("Use item[%r] to get field value" % name)
AttributeError: Use item['type'] to get field value
```

**Comment by elacuesta** [comment]
> **Expected behavior:** `self['field_name']` should return the `Field()` attribute.

`self['field_name']` is expected to [return the _value_ of the field](https://github.com/scrapy/scrapy/blob/1.8/scrapy/item.py#L90-L91), not the _field itself_.

___

This doesn't look like a bug to me, more like a design choice. The internal `_values` dictionary is populated only after setting a specific value, like in the following example:

```python
In [1]: from scrapy import Item, Field
   ...:
   ...: class CustomItem(Item):
   ...:     foo = Field()
   ...:
   ...: i = CustomItem()

In [2]: i["foo"]
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-8412772f5c45> in <module>
----> 1 i["foo"]

~/.../scrapy/item.py in __getitem__(self, key)
     89
     90     def __getitem__(self, key):
---> 91         return self._values[key]
     92
     93     def __setitem__(self, key, value):

KeyError: 'foo'

In [3]: i["foo"] = "bar"

In [4]: i["foo"]
Out[4]: 'bar'
```

Note that this also happens in the [`Item.__init__` method](https://github.com/scrapy/scrapy/blob/1.8/scrapy/item.py#L87-L88), if the object receives values when it's created:

```python
In [1]: from scrapy import Item, Field
   ...:
   ...: class BaseItem(Item):
   ...:     url=Field()
   ...:     id=Field()
   ...:     type=Field()
   ...:
   ...:     def __init__(self, *args, **kwargs):
   ...:         super().__init__(*args, **kwargs)
   ...:         if not self['type']: #<--- error here. Raises KeyError
   ...:             raise NotImplementedError('type of ad must be set as static class variable')
   ...:         #use url hash as id to avoid duplicates
   ...:         self['id']=hash(self.get('url'))
   ...:

In [2]: b = BaseItem(url="https://example.org", type="foobar")

In [3]: b
Out[3]: {'id': -872504314545110605, 'type': 'foobar', 'url': 'https://example.org'}
```
___

**TL;DR**
Not sure if it fits the specific use case, but the `KeyError` could be avoided by using `get` instead of `__getitem__`, i.e. `if not self.get('type'):` instead of `if not self['type']:`
